### ISSUE TITLE
plpgsql: IF statement is not supported

### ISSUE THREAD

**Issue by giangpham712** [cockroachdb/cockroach#108700]
The following SQL doesn't work

```
IF NOT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname = 'test') THEN
    CREATE SCHEMA test;
END IF;
```

It returns error `at or near "if": syntax error`

**Additional context**
This affects efcore.pg tests

These test use this script to create the schema conditionally because

```
PostgreSQL has CREATE SCHEMA IF NOT EXISTS, but that requires CREATE privileges on the database even if the schema already exists. This blocks multi-tenant scenarios where the user has no database privileges. So we procedurally check if the schema exists instead, and create it if not.
```

If CockroachDB can allow the schema existence check to be performed without database privileges and only requires the privileges for executing the CREATE then we can use `CREATE SCHEMA IF NOT EXISTS` for CockroachDB

@fqazi 

Jira issue: CRDB-30603

**Comment by fqazi** [comment]
@giangpham712 Please skip this test for now, since this won't be fixed any time soon. As a workaround, you can use IF NOT EXISTS safely

**Comment by mgartner** [comment]
The example is PL/pgSQL, not SQL, correct? PLpgSQL is only allowed with user-defined procedures or procedures. I'll close this for now.

**Comment by rafiss** [comment]
@mgartner this query was generated by EFCore and @giangpham712 is a contractor working with us. I am assuming that the tool is indeed sending this in a plpgsql query. Is there another issue to track adding this syntax? I will reopen this unless there is another one.

**Comment by mgartner** [comment]
We've already merged support for `IF THEN ... ; END IF;` statements in PLpgSQL UDFs and procedures, so it will be available in v23.2. However, we don't support DDL statements like `CREATE SCHEMA ...` in UDFs procedures, and it's not on our near-term road map. I don't think we had an issue tracking this, so I created #110080.

**Comment by rafiss** [comment]
OK thanks for explaining! in that case, we can keep this closed. @giangpham712 , the CRDB support for the IF syntax will arrive in 23.2, but will not be backported to 23.1. So feel free to skip the tests that require it.
